front/app/containers/IdeasFeedPage/BottomSheet.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useState, useRef, useCallback } from 'react';
+
+import { Box, colors } from '@citizenlab/cl2-component-library';
+import styled from 'styled-components';
+
+const COLLAPSED_HEIGHT = 60;
+
+const SHEET_HEIGHT = 'calc(80vh - 60px)';
+
+const Container = styled.div<{ translateY: number; isDragging: boolean }>`
+  position: fixed;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  background: ${colors.white};
+  border-top-left-radius: 16px;
+  border-top-right-radius: 16px;
+  box-shadow: 0 -4px 20px rgba(0, 0, 0, 0.15);
+  transform: translateY(${({ translateY }) => translateY}px);
+  transition: ${({ isDragging }) =>
+    isDragging ? 'none' : 'transform 0.3s ease-out'};
+  height: ${SHEET_HEIGHT};
+  z-index: 1020;
+`;
+
+const DragHandle = styled.div`
+  width: 40px;
+  height: 4px;
+  background: ${colors.grey400};
+  border-radius: 2px;
+  margin: 12px auto;
+`;
+
+const DragArea = styled.button`
+  display: block;
+  width: 100%;
+  padding: 8px 0;
+  touch-action: none;
+  background: none;
+  border: none;
+  cursor: grab;
+
+  &:active {
+    cursor: grabbing;
+  }
+`;
+
+interface Props {
+  children: React.ReactNode;
+  a11y_panelLabel: string;
+  a11y_expandLabel: string;
+  a11y_collapseLabel: string;
+}
+
+const BottomSheet = ({
```

Comment:
This is built in a way where I plan to convert it to a reusable component. I suspect we will need this type of interface for in the future. I am surprised we did not already have something like this. I want to play with it a bit more here before moving it to our reusable components, though.